matlab/src/cpp/arrow/matlab/array/proxy/chunked_array.cc

```diff
@@ -0,0 +1,187 @@
+        for (const auto& array_proxy_id : array_proxy_ids) {
+            auto proxy = libmexclass::proxy::ProxyManager::getProxy(array_proxy_id);
+            auto array_proxy = std::static_pointer_cast<proxy::Array>(proxy);
+            auto array = array_proxy->unwrap();
+            arrays.push_back(array);
+        }
+
+        auto proxy = libmexclass::proxy::ProxyManager::getProxy(type_proxy_id[0]);
+        auto type_proxy = std::static_pointer_cast<type::proxy::Type>(proxy);
+        auto type = type_proxy->unwrap();
+
+        MATLAB_ASSIGN_OR_ERROR(auto chunked_array, 
+                               arrow::ChunkedArray::Make(arrays, type),
+                               error::CHUNKED_ARRAY_MAKE_FAILED);
+
+        return std::make_unique<proxy::ChunkedArray>(std::move(chunked_array));
+    }
+
+    std::shared_ptr<arrow::ChunkedArray> ChunkedArray::unwrap() {
+        return chunked_array;
+    }
+
+    void ChunkedArray::getLength(libmexclass::proxy::method::Context& context) {
+        namespace mda = ::matlab::data;
+        mda::ArrayFactory factory;
+        auto length_mda = factory.createScalar(chunked_array->length());
+        context.outputs[0] = length_mda;
+    }
+
+    void ChunkedArray::getNumChunks(libmexclass::proxy::method::Context& context) {
+        namespace mda = ::matlab::data;
+        mda::ArrayFactory factory;
+        auto length_mda = factory.createScalar(chunked_array->num_chunks());
+        context.outputs[0] = length_mda;
+    }
+
+    void ChunkedArray::getChunk(libmexclass::proxy::method::Context& context) {
+        namespace mda = ::matlab::data;
+        mda::ArrayFactory factory;
+        
+        mda::StructArray args = context.inputs[0];
+        const mda::TypedArray<int32_t> index_mda = args[0]["Index"];
+        const auto matlab_index = int32_t(index_mda[0]);
+        
+        // Note: MATLAB uses 1-based indexing, so subtract 1.
+        // arrow::Schema::field does not do any bounds checking.
+        const int32_t index = matlab_index - 1;
+        const auto num_chunks = chunked_array->num_chunks();
+        
+        if (num_chunks == 0) {
+            context.error = makeEmptyChunkedArrayError();
+            return;
+        }
+        
+        if (matlab_index < 1 || matlab_index > num_chunks) {
+            context.error = makeInvalidNumericIndexError(matlab_index, num_chunks);
+            return;
+        }
+
+        const auto array = chunked_array->chunk(index);
+        MATLAB_ASSIGN_OR_ERROR_WITH_CONTEXT(auto array_proxy,
+                                            arrow::matlab::array::proxy::wrap(array),
+                                            context,
+                                            error::UNKNOWN_PROXY_FOR_ARRAY_TYPE);
+        
+        
+        const auto array_proxy_id = libmexclass::proxy::ProxyManager::manageProxy(array_proxy);
+        const auto type_id = static_cast<int64_t>(array->type_id());
```

Comment:
This uses `int64_t` for `arrow::Type::type` and https://github.com/apache/arrow/pull/37525/files#diff-372554407460398c1974859c73ab50b2bbae13bd3a6826a6097981a07924d534R155 uses `int32_t` for `arrow::Type::type`. Which is preferred?

Reply:
Ah, I meant to use `int32_t` here. I'll update the code to use `int32_t`.